ntp-proto/src/peer.rs

```diff
@@ -198,23 +199,20 @@ impl Peer {
         self.last_poll_interval
     }
 
-    pub fn generate_poll_message(&mut self, local_clock_time: NtpInstant) -> NtpHeader {
+    pub fn generate_poll_message(&mut self) -> NtpHeader {
         self.reach.poll();
 
         let mut packet = NtpHeader::new();
         packet.poll = self.last_poll_interval.as_log();
         packet.mode = NtpAssociationMode::Client;
 
-        // we write into the origin_timestamp and transmit_timestamp to validate the packet we get
-        // back. The origin_timestamp must not be changed, the transmit_timestamp must be changed
-        let local_clock_time = NtpTimestamp::from_bits(local_clock_time.to_bits());
-        self.next_expected_origin = Some(local_clock_time);
-
-        // in handle_incoming, we check that this field is unchanged
-        packet.origin_timestamp = local_clock_time;
-
-        // in handle_incoming, we check that this field was updated
-        packet.transmit_timestamp = local_clock_time;
+        // In order to increase the entropy of the transmit timestamp
+        // it is just a randomly generated timestamp.
+        // We then expect to get it back identically from the remote
+        // in the origin field.
+        let transmit_timestamp = thread_rng().gen();
+        self.next_expected_origin = Some(transmit_timestamp);
```

Comment:
I think we discussed earlier that we expect a `self.next_expected_origin = None` somewhere. Is that not needed anymore (my impression is that the value is overwritten in time. But maybe we should clear it just to be explicit?)

Reply:
Here it should be set, as generate_poll_message indicates we are starting the next poll cycle, and we should update next_expected_origin to an actual value regardless of what it was (if it still had a value here it just indicates a missed poll interval, which is handled already in a different way by self.reach)